# Experiment Validation

This notebook evaluates whether the A/B test satisfies the assumptions
required for valid causal inference.

In [ ]:
import pandas as pd

df = pd.read_csv("../data/processed/ab_data_cleaned.csv")

df.head()

# Sample Ratio Mismatch (SRM)

We assess whether the observed allocation between control and treatment
groups is consistent with the expected 50/50 split.

A significant deviation may indicate issues in randomization, data
collection, or experiment execution, potentially invalidating causal inference.

In [11]:

from scipy.stats import chisquare

observed = df["group"].value_counts().reindex(["control", "treatment"])
observed

group
control      146926
treatment    147552
Name: count, dtype: int64

The observed counts differ slightly between groups, which is expected due to
random variation. We now test whether this difference is statistically significant.

To formally evaluate this, we use a chi-square goodness-of-fit test.

The null hypothesis assumes that the observed group proportions follow the
expected allocation:

$H_0$: The observed frequencies match the expected 50/50 distribution

$H_1$: The observed frequencies deviate from the expected distribution

In [ ]:
expected = pd.Series(
    [len(df) / 2, len(df) / 2],
    index=["control", "treatment"]
)

expected # 50/50 distribution

control      147239.0
treatment    147239.0
dtype: float64

In [16]:
chi2_stat, p_value = chisquare(f_obs=observed, f_exp=expected)
chi2_stat, p_value

(np.float64(1.3307479675901086), np.float64(0.2486721971710374))

The p-value (0.2487) is substantially higher than the typical significance
level (e.g., 0.05), so we fail to reject the null hypothesis. This indicates 
that the observed allocation is consistent with the expected 50/50 split, and 
no evidence of sample ratio mismatch (SRM) is found.

This supports the validity of the experiment's randomization process.